src/source/spec_loader.rs

```diff
@@ -0,0 +1,121 @@
+#[cfg(feature = "spec")]
+use std::{io::Cursor, path::PathBuf};
+
+#[cfg(feature = "spec")]
+use bevy::asset::{AssetLoader, LoadContext, LoadedAsset};
+#[cfg(feature = "spec")]
+use bevy_utils::BoxedFuture;
+#[cfg(feature = "spec")]
+use kira::sound::{error::SoundFromFileError, Sound, SoundSettings};
+#[cfg(feature = "spec")]
+use serde::Deserialize;
+
+#[cfg(feature = "spec")]
+use crate::AudioSource;
+
+#[derive(Default)]
+pub struct SpecLoader;
+
+/// Kira sound settigns specification
+///
+/// This is used when loading from a *.{wav,mp3,ogg,flac}.ron file to override
+/// the default `kira::SoundSettings`.
+#[cfg(feature = "spec")]
+#[derive(Deserialize)]
+struct SoundSettingsSpec {
```

Comment:
Could we use `f64` here instead of `Option<f64>`? I think it would make the Iron asset nice to write/read. If a field is not set, we just pass a `None`.

Reply:
Since these values must be positive to be meaningful, I've made it so that the struct takes `f64`s and if they are negative, converts them to `None` when constructing `SoundSettings`.